TCAT/Controllers/FavoritesViewController.swift

```diff
@@ -12,19 +12,43 @@ import UIKit
 class FavoritesViewController: UIViewController {
 
     // MARK: - View vars
+    private let collapsedRevealHeight: CGFloat = 54
     private let editButton = UIButton()
     private var favoritesCollectionView: UICollectionView!
     private let favoritesTitleLabel = UILabel()
+    private let partialRevealHeight: CGFloat = 192
     private let tabView = UIView()
     private let tabSize = CGSize(width: 32, height: 4)
 
     // MARK: - Data vars
     private let favoritesReuseIdentifier = "FavoritesCollectionViewCell"
-    private var favorites: [String] = ["Collegetown Bagels", "Collegetown Bagels", "Collegetown Bagels", "Collegetown Bagels"] // Temp
+    private let addFavoritesReuseIdentifier = "AddFavoritesCollectionViewCell"
+    private var favoritePlaces: [Place] {
+        didSet {
+            favoritesCollectionView.reloadData()
+        }
+    }
+    private var isEditingFavorites: Bool {
+        didSet {
+            setupButtons()
```

Comment:
i noticed we toggle `isEditingFavorites` which means we add `editButton` multiple times. is this necessary or should we be removing them as we go as well?

Reply:
can you explain this comment a bit more? my intent is to have isEditingFavorites keep track of our state so that we know what viewing mode we are in. So every time our state changes, I wanted to reset up our buttons and reload our collection view.

Reply:
Yeah I might be misunderstanding it but from what it seems like, each call to `setupButtons()` creates a new button and adds it as a subview and its possible that it gets called multiple times. I don't see where we're ever removing these buttons as subviews which means we are just adding more and more buttons onto the screen which seems unnecessary?